apps/examples/wifi_manager/wm_test/wm_test_interop.c

```diff
@@ -0,0 +1,665 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License\n");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <wifi_manager/wifi_manager.h>
+#include "wm_test.h"
+#include "wm_test_log.h"
+
+static sem_t g_wio_sem;
+
+#define TAG "[WTIO]"
+
+#define DELIMITER ","
+#define DELI_LEN 1
+#define WIFI_INTEROP_BUFSIZE 128
+#define WIFI_INTEROP_PATH "/mnt/"
+#define WIFI_INTEROP_FILENAME "wm_aplist.conf"
+#define WIFI_INTEROP_RESULT "wm_result.conf"
+
+#define WIO_TEST_SIGNAL								\
+	do {											\
+		sem_post(&g_wio_sem);			\
+		WT_LOG(TAG, "send signal");	\
+	} while (0)
+
+#define WIO_TEST_WAIT								\
+	do {											\
+		WT_LOG(TAG, "wait signal");	\
+		sem_wait(&g_wio_sem);				\
+	} while (0)
+
+
+
+typedef enum {
+	SSID,
+	SECURITY,
+	PASSPHRASE
+} wio_config_e;
+
+static int g_ap_found_status;
+static int g_current_ap_idx;
+static int g_connect_status;
+
+#define MAX_INTEROP_AP_CONFIG_COUNT 10
+#define MAX_INTEROP_SECURITY_LEN 10
+
+typedef struct {
+	wifi_manager_ap_config_s ap_config;
+	char security[MAX_INTEROP_SECURITY_LEN];
+	int scan_result;
+	int connect_result;
+} interop_ap_info_s;
+
+typedef struct {
+	int ap_count;
+	interop_ap_info_s ap_info[MAX_INTEROP_AP_CONFIG_COUNT];
+} interop_ap_config_list_s;
+
+static interop_ap_config_list_s g_apconfig_list;
+
+
+static void print_wifi_ap_profile(wifi_manager_ap_config_s *config)
```

Comment:
please add _prefix all internal functions for readability in this program(wm_test)